eclass/gnome2.eclass

```diff
@@ -23,7 +23,7 @@ case "${EAPI:-0}" in
 	4|5)
 		EXPORT_FUNCTIONS src_unpack src_prepare src_configure src_compile src_install pkg_preinst pkg_postinst pkg_postrm
 		;;
-	6)
+	6|7)
```

Comment:
Here I would like to take a chance to fix a few non-EAPI related things in the same push (some of them might end up as EAPI-7 only behaviour change), but that can be a follow-up to when we are done with this patchset - just please allow me to finish that work and push it all in at once (and feel free to keep reminding me on IRC to not delay EAPI-7 enablement of all this)

Reply:
We'll see how things are once things are reworked. Might be fine to push, but might be some things that need changing together with EAPI-7 for gnome.eclass. I'll try to find out while the patchset gets reworked.

Reply:
as-is with current commits, gnome2.eclass does not work with EAPI-7. eutils and prune_libtool_files are still there, and those do not, nor will work, with EAPI-7.

Reply:
@leio I can remove the EAPI7 support stuff from gnome2.eclass; the only reason it and
gnome2-utils.eclass are being touched really is the transplant of the icon code to the
xdg*.eclass'es and figured I'd handle that as well. On another note, a large amount of
gnome packages are moving to meson, which would require a whole retooling of the
basic functions, so perhaps we should work with upstream to port them moving forward
and have a new eclass for gnome stuff, or just do meson.eclass + gnome2-utils.eclass
as needed.

Reply:
gnome2.eclass won't be used for meson based builds.
If we could move `gnome2_schemas_update` and co to xdg too, I wouldn't even need anything other than `inherit gnome.org meson xdg` most of the time, just some extra stuff for the few packages that ship GIO modules, pixbuf loaders and the likes.
GNOME 3.30 still has some packages as autotools only, and with some of those I don't expect it to change for 4-12 months either. So we'll eventually need gnome2.eclass supporting EAPI-7.

If really needed, I can be fine with just a note in there to kill the fallbacks with EAPI-7, I suppose - or only introduce EAPI-7 support for gnome2-utils and not gnome.eclass for now. Though the request was to add support for all of them ;)  But I can take what I can get :)

Reply:
Personally I'd feel better keeping that stuff in gnome2-utils, since they are pretty
distinct and bringing all that into non-gnome packages which have desktop files,
icons, and mimeinfo seems overkill.

Reply:
gnome2_schemas_update does not bring any new dependencies, only a "find" call that most other distributions runs for EVERY package with triggers, not just those that pull in some certain packaging helpers. It's also needed for any glib gsettings using project, not just gnome, but all up to snuff gtk programs, other desktops and even some glib-based CLI programs.
That said, as talked on IRC, if no-one has the time to immediately discuss this over on gentoo-dev and the like, we can just postpone this for a separate potential change.